vllm/compilation/backends.py

```diff
@@ -148,9 +157,113 @@ def fix_functionalization(graph: fx.Graph):
     #     print(graph.python_code(root_module="self", verbose=True).src, file=f)
 
 
-def vllm_backend(graph, example_inputs):
+def wrap_inductor(graph, example_inputs, additional_inductor_config):
     from torch._inductor import config
     current_config = config.shallow_copy_dict()
     from torch._inductor.compile_fx import compile_fx
+
+    if additional_inductor_config is not None:
+        current_config.update(additional_inductor_config)
+    if current_config['post_grad_custom_post_pass'] is not None:
+        logger.warning(
+            "post_grad_custom_post_pass is already set in the config. "
+            "Overwriting it with the fix_functionalization")
```

Comment:
I think it will be common to add custom passes - I think we should handle this by wrapping the provided pass instead of overwriting:

```suggestion
    if additional_inductor_config is not None:
        current_config.update(additional_inductor_config)
    def noop_pass(graph: fx.Graph):
        pass
    
    old_post_pass = current_config['post_grad_custom_post_pass'] or noop_pass

    def post_grad_custom_post_pass(graph: fx.Graph):
        old_pass(graph)
        fix_functionalization(graph)
   current_config['post_grad_custom_post_pass'] = post_grad_custom_post_pass
```